GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -34,12 +34,41 @@ public GVFSVerb(bool validateOrigin = true)
         public abstract string EnlistmentRootPathParameter { get; set; }
 
         [Option(
-            GVFSConstants.VerbParameters.Mount.ServiceName,
-            Default = GVFSConstants.Service.ServiceName,
+            GVFSConstants.VerbParameters.Mount.InternalParameters,
             Required = false,
-            HelpText = "This parameter is reserved for internal use.")]
+            HelpText = "This parameter is reserved for internal use." +
+                " Accepts input in the form key1=value;key2=value;key3=value etc..")]
+        public string InternalParameters
+        {
+            set
+            {
+                if (!string.IsNullOrEmpty(value))
+                {
+                    string[] values = value.Split(';');
+                    foreach (string s in values)
+                    {
+                        string[] pair = s.Split('=');
+                        if (pair.Length == 2)
+                        {
+                            if (string.Equals(pair[0], GVFSConstants.VerbParameters.Mount.StartedByService))
+                            {
+                                this.StartedByService = pair[1];
+                            }
+
+                            if (string.Equals(pair[0], GVFSConstants.VerbParameters.Mount.ServiceName))
+                            {
+                                this.ServiceName = pair[1];
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
         public string ServiceName { get; set; }
 
+        public string StartedByService { get; set; }

```

Comment:
This should be a bool, not a string. But this is moot if you parse the json into a strongly-typed `InternalParameters` instance.